stestr/repository/file.py

```diff
@@ -19,12 +19,22 @@
 import sys
 import tempfile
 
-from six.moves import dbm_gnu as dbm
+import six
 from subunit import TestProtocolClient
 import subunit.v2
 import testtools
 from testtools.compat import _b
 
+if sys.platform != 'win32':
+    from six.moves import dbm_gnu as dbm
+else:
+    # TODO(claudiub): replace import with six.moves.dbm once the dbm move is
+    # defined: https://github.com/benjaminp/six/issues/179
+    if six.PY2:
+        import anydbm as dbm
```

Comment:
So I just checked this and the bug I thought was there will hit on windows if you use anydbm.  Python2 will use dbhash which no longer exists in python3. (python3 on windows defaults to dbm.dumb) So if you create a repository using python2 first it will crash on python3 when you try to use that same repository.

The only way around this I see is to use dbm.dumb explicitly in both cases. I wonder how big the performance hit is if we used it for every os: https://docs.python.org/3/library/dbm.html#module-dbm.dumb